core/src/main/java/novoda/lib/sqliteprovider/util/DBUtils.java

```diff
@@ -162,4 +175,42 @@ public static String getSQLiteVersion() {
         indexCursor.close();
         return constraints;
     }
+
+    private static Constraint findIntegerPrimaryKeyConstraint(SQLiteDatabase database, String table) {
+        final Cursor cursor = queryTableColumnsFor(table, database);
+        try {
+            while (cursor.moveToNext()) {
+                if (isTableInfoItemAnIntegerPrimaryKey(cursor)) {
+                    String columnName = cursor.getString(cursor.getColumnIndex(COLUMN_NAME));
+                    return new Constraint(Collections.singletonList(columnName));
+                }
+            }
+        } finally {
+            cursor.close();
+        }
+        return null;
+    }
+
+    private static Cursor queryTableColumnsFor(String table, SQLiteDatabase database) {
+        return database.rawQuery(String.format(PRAGMA_TABLE_INFO, table), null);
+    }
+
+    private static Cursor queryIndexListForTable(String table, SQLiteDatabase database) {
+        return database.rawQuery(String.format(PRGAMA_INDEX_LIST, table), null);
+    }
+
+    private static Cursor queryIndexInfo(String index, SQLiteDatabase database) {
+        return database.rawQuery(String.format(PRGAMA_INDEX_INFO, index), null);
+    }
+
+    private static boolean isTableInfoItemAnIntegerPrimaryKey(Cursor cursor) {
+        int pkInt = cursor.getInt(cursor.getColumnIndex(COLUMN_PRIMARY_KEY));
+        String columnType = cursor.getString(cursor.getColumnIndex(COLUMN_TYPE));
+
+        boolean isPrimaryKey = pkInt == 1;
```

Comment:
according to https://www.sqlite.org/pragma.html#pragma_table_info the check should be `pkInt != 0`